packages/skill-api/src/core/services/convertkit/convertkit-tag-purchase.ts

```diff
@@ -46,57 +44,62 @@ async function getSanityProduct(productId: string) {
 
 export async function convertkitTagPurchase(email: string, purchase: Purchase) {
   // skip CK tagging in development
-  if (process.env.SKIP_CK_TAGGING === 'true') return
+  if (process.env.SKIP_CK_TAGGING === 'true')
+    return {tagged: false, reason: 'SKIP_CK_TAGGING is true'}
 
   try {
     const sanityProduct = await getSanityProduct(purchase.productId)
 
-    const convertkitPurchasedTagId =
+    const convertkitPurchasedTagId: string | undefined =
       sanityProduct?.convertkitPurchasedTagId ||
       process.env.CONVERTKIT_PURCHASED_TAG_ID
 
-    let subscriber
-
-    if (sanityProduct?.convertkitPurchasedTagId) {
-      subscriber = await tagSubscriber(email, convertkitPurchasedTagId)
-    }
+    if (!convertkitPurchasedTagId) {
+      if (process.env.NODE_ENV === 'production') {
+        await postToSlack({
+          webClient: new WebClient(process.env.SLACK_TOKEN),
+          channel: process.env.SLACK_ANNOUNCE_CHANNEL_ID,
+          text: `process.env.CONVERTKIT_PURCHASED_TAG_ID is not set for ${purchase.productId}`,
+          attachments: [],
+        })
+      }
 
-    if (process.env.CONVERTKIT_PURCHASED_TAG_ID) {
-      subscriber = await tagSubscriber(email, convertkitPurchasedTagId)
+      // no tagging is going to happen (unless we want to lookup by email), so return early
+      return {
+        tagged: false,
+        reason: 'convertkitPurchasedTagId is not set',
+      }
```

Comment:
if we don't have a purchased tag id, then post to slack (in production) and then return early.

unless we want to add new logic to look up the subscriber by `email` and still try to do the field tagging with that?